### PR TITLE
plots: add smooth template

### LINKED ISSUE #3906: plots smoothing
Our default plot is not very informative when the number of data points is huge (1k):
visualization_

We need to introduce some smoothing into the default plot or create a new smoothing template. The result should look like:
visualization (2)

PS: I've done the last plot by adding `loess` transformation to the template which breaks the revision. We need to fix the revision or find another solution. 
```
    "transform": [{
        "loess": "<DVC_METRIC_Y>",
        "on": "<DVC_METRIC_X>"
    }
    ],
```

### PR DESCRIPTION
**Pull request by pared**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here. If the CLI API is changed, I have updated [tab completion scripts](https://github.com/iterative/dvc/tree/master/scripts/completion).

* [x] ❌ I will check DeepSource, CodeClimate, and other sanity checks below. (We consider them recommendatory and don't expect everything to be addressed. Please fix things that actually improve code or fix bugs.)

Thank you for the contribution - we'll try to review it as soon as possible. 🙏
Fixes #3906

### CODE REVIEW

File: dvc/repo/plots/template.py [hunk 1]
```diff
@@ -168,12 +168,46 @@ class DefaultScatterTemplate(Template):
...
+                "scale": {"zero": False},
+            },
+            "color": {"field": "rev", "type": "nominal"},
+        },
+        "transform": [
+            {
+                "loess": Template.Y_ANCHOR,
+                "on": Template.X_ANCHOR,
+                "groupby": ["rev"],
+                "bandwidth": 0.3,
```
Comment (pared) [inline dvc/repo/plots/template.py:198]: Added so that users will not have to search for parameter name

**Comment by efiop** [comment]
Wonder if it is worth adding a smooth example to https://dvc.org/doc/command-reference/plots ?

**Comment by dmpetrov** [comment]
@efiop for sure, the example is needed

**Comment by efiop** [comment]
@dmpetrov Looks like smoothing doesn't change anything in the examples we have in https://dvc.org/doc/command-reference/plots . @pared @dmpetrov could you guys provide an example for https://dvc.org/doc/command-reference/plots , please?

**Comment by jorgeorpinel** [comment]
Hi. Docs is checked but I don't see any ref. to an issue or PR in dvc.org 😢

**Comment by efiop** [comment]
@jorgeorpinel Created https://github.com/iterative/dvc.org/issues/1439 so we don't forget about it.

**Comment by jorgeorpinel** [comment]
Much appreciated!

**Pull request merged**